pkg/snapshotter/snapshotter.go

```diff
@@ -76,6 +77,12 @@ type Driver interface {
 	RestoreStatus(pvcName, namespace string) (RestoreInfo, error)
 	// CancelRestore cancels a restore operation
 	CancelRestore(pvcName, namespace string) error
+	// Upload objects to cloud
+	UploadSnapshotData(backupLocation *storkapi.BackupLocation, snapshotInfo SnapshotInfo, snapName, objectPath, objectName string) error
+	// Download objects from cloud
+	DownloadObject(backupLocation *storkapi.BackupLocation, objectPath string) ([]byte, error)
```

Comment:
Instead of returning the []byte array, you should create the SnapshotInfo object here and return that. Let the callers of this API do the translation of SnapshotInfo object to respective CSI objects.